app/src/main/java/pm/gnosis/heimdall/ui/safe/overview/SafeAdapter.kt

```diff
@@ -72,28 +80,62 @@ class SafeAdapter @Inject constructor(
         abstract fun castedBind(data: T, payloads: List<Any>?)
     }
 
-    inner class ViewHolder(itemView: View) : CastingViewHolder<Safe>(Safe::class.java, itemView), View.OnClickListener {
+    inner class ViewHolder(itemView: View) : CastingViewHolder<SafeWithInfo>(SafeWithInfo::class.java, itemView), View.OnClickListener {
+        private val disposables = CompositeDisposable()
 
         init {
             itemView.setOnClickListener(this)
             itemView.layout_safe_item_share.setOnClickListener {
                 currentEntry?.let {
                     it.let {
-                        val addressString = it.address.asEthereumAddressString()
-                        val title = context.getString(R.string.sharing_x, it.name ?: addressString)
-                        context.shareExternalText(addressString, title)
+                        val addressString = it.safe.address.asEthereumAddressString()
+                        shareSelection.onNext(addressString)
                     }
                 }
             }
         }
 
-        private var currentEntry: Safe? = null
+        private var currentEntry: SafeWithInfo? = null
 
-        override fun castedBind(data: Safe, payloads: List<Any>?) {
+        override fun castedBind(data: SafeWithInfo, payloads: List<Any>?) {
             currentEntry = data
-            itemView.layout_safe_item_address.text = data.address.asEthereumAddressString()
-            itemView.layout_safe_item_name.text = data.name
-            itemView.layout_safe_item_name.visibility = if (data.name.isNullOrEmpty()) View.GONE else View.VISIBLE
+            itemView.layout_safe_item_address.text = data.safe.address.asEthereumAddressString()
+            itemView.layout_safe_item_name.text = data.safe.name
+            itemView.layout_safe_item_name.visibility = if (data.safe.name.isNullOrEmpty()) View.GONE else View.VISIBLE
+            if (data.info != null) {
+                itemView.layout_safe_item_authorizations.text = "${data.info.requiredConfirmations}/${data.info.owners.count()}"
+                itemView.layout_safe_item_ether.text = data.info.balance.toEther().stringWithNoTrailingZeroes()
+
+                itemView.layout_safe_item_owner.visibility =
+                        if (data.info.owners.map { it.hexAsBigIntegerOrNull() }.any { accountAddress == it }) View.VISIBLE
+                        else View.GONE
+            } else {
+                itemView.layout_safe_item_authorizations.text = "-"
+                itemView.layout_safe_item_ether.text = "-"
+            }
+        }
+
+        @OnLifecycleEvent(Lifecycle.Event.ON_START)
+        fun start() {
+            // Make sure no disposable are left over
+            disposables.clear()
+            if (currentEntry?.info == null) {
+                currentEntry?.safe?.address?.let { address ->
+                    disposables += safeRepository.loadInfo(address)
```

Comment:
Shouldn't we use a viewmodel here?

I know we talked about it, but maybe it makes sense to implement a simple cache in the viewmodel instead of the viewholder. So the viewmodel could have a HashMap<BigInteger, Info>.  Not sure if that might be too much work for now. Or we directly do it in the repo and use shared observables (there is an exension method for that)